bedrock-codec/src/main/java/org/cloudburstmc/protocol/bedrock/codec/v291/BedrockCodecHelper_v291.java

```diff
@@ -383,4 +386,18 @@ public <T> void writeOptional(ByteBuf buffer, Predicate<T> isPresent, T object,
             consumer.accept(buffer, object);
         }
     }
+
+    @Override
+    public PlayerPermission readPlayerPermission(final int ordinal) {
```

Comment:
Do we really need a helper method specifically for this? If anything, it would make more sense to have a generic helper method for all enums that we want to fail silently. 

Also not sure how I feel about returning null since users probably expect a value to be returned no matter what. This will also cause packet encoding to be asymmetric since decoding any out-of-bounds enum value will get encoded to -1.

Reply:
I think using helper method is a good solution to reduce duplicated code, if this does not match the style of this project, i'll replace it.

For the out-of-bounds case, i have an idea of replace PlayerPermission with non-enum class and accepting custom values. Because i found mineplex sending value `223` for PlayerPermission in `PlayerAbilities`, what's your opinion?